addons/binding/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/internal/ssl/PinTrustManager.java

```diff
@@ -133,32 +112,29 @@ public void checkServerTrusted(X509Certificate @Nullable [] chainN, @Nullable St
         }
     }
 
-    @Override
-    public X509Certificate[] getAcceptedIssuers() {
-        return new X509Certificate[0];
-    }
-
-    @Override
-    public void checkClientTrusted(X509Certificate @Nullable [] chain, @Nullable String authType,
-            @Nullable Socket socket) throws CertificateException {
-        checkClientTrusted(chain, authType);
-    }
-
-    @Override
-    public void checkClientTrusted(X509Certificate @Nullable [] chain, @Nullable String authType,
-            @Nullable SSLEngine sslEngine) throws CertificateException {
-        checkClientTrusted(chain, authType);
+    protected byte[] getEncoded(PinType type, X509Certificate cert) throws CertificateEncodingException {
+        switch (type) {
+            case CERTIFICATE_TYPE:
+                return cert.getEncoded();
+            case PUBLIC_KEY_TYPE:
+                return cert.getPublicKey().getEncoded();
+        }
+        throw new CertificateEncodingException("Type unknown");
     }
 
-    @Override
-    public void checkServerTrusted(X509Certificate @Nullable [] chain, @Nullable String authType,
-            @Nullable Socket socket) throws CertificateException {
-        checkServerTrusted(chain, authType);
+    /**
+     * A signature name depends on the security provider but usually follows
+     * https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#Signature.
+     * E.g.: "SHA256withRSA". We need "SHA" and "256" to initialize a {@link PinMessageDigest}.
+     */
+    PinMessageDigest getMessageDigestForSigAlg(String sigAlg) throws CertificateException {
+        final Matcher matcher = Pattern.compile("(\\D*)(\\d+)").matcher(sigAlg);
+        matcher.find();
+        try {
+            return new PinMessageDigest(matcher.group(1) + "-" + matcher.group(2));
```

Comment:
This is two lines shorter but nowbody knowns what group1 and group2 stands for like in the previous version.